the-collab-lab/tcl-75-smart-shopping-list
# [Issue 10]: Empty/duplicate item alerts

## Description

This code shows a series of errors if the user tries to submit an empty item, or an item that is identical to an existing item name. Additionally, if the user enters an item name with any spacing or punctuation, those pieces are removed to normalize the casing. The users original input is saved to the database. 

## Related Issue

closes #10 

## Acceptance Criteria


## Type of Changes

enhancement

## Updates

### Before
![Screenshot 2024-09-10 at 10 48 23 AM](https://github.com/user-attachments/assets/45eb5ce5-d2eb-4c01-a4f9-84962c5e6d5e)
![Screenshot 2024-09-10 at 10 49 15 AM](https://github.com/user-attachments/assets/962b94d6-17a3-42b6-86a3-7ec1bd785853)
![Screenshot 2024-09-10 at 10 50 02 AM](https://github.com/user-attachments/assets/4db81721-9ccd-48bf-b474-defa3480e5de)

### After
![Screenshot 2024-09-10 at 10 57 45 AM](https://github.com/user-attachments/assets/81a6b770-4432-40b9-87b3-6c72a62f5799)
![Screenshot 2024-09-10 at 10 58 20 AM](https://github.com/user-attachments/assets/fbb2a160-fd01-4a34-bc3d-ad7b7cbb3772)
![Screenshot 2024-09-10 at 10 58 38 AM](https://github.com/user-attachments/assets/b4ad32e1-fdd9-44df-b3aa-c0317a761994)
![Screenshot 2024-09-10 at 11 07 12 AM](https://github.com/user-attachments/assets/489c1451-3a5c-4cf8-8e25-a0d35cce9141)

## Testing Steps / QA Criteria

1. Sign into the app and choose a list or add a new one
2. Enter item that is identical to current item on the list
3. Enter item with punctuation and/or different casing from a current list item
4. Check that user's original input is saved in the database
